### DISCUSSION TITLE
How to use mjd_transitionFD in python binding

### DISCUSSION THREAD

**Discussion by yzd0014** [google-deepmind/mujoco#1076] [General]
My questions has two parts, first when mjd_transitionFD is called, I don't need to call mj_step anymore, is this right?

Second, when mjd_transitionFD is used, I got an error, I don't know how to fix this error.

<img width="593" alt="image" src="https://github.com/google-deepmind/mujoco/assets/10913570/8135f21c-6c7a-4ed5-93ef-a4eecfcd1bf7">

<img width="1823" alt="image" src="https://github.com/google-deepmind/mujoco/assets/10913570/94cf0bee-e9b3-44f7-b864-1b9e305dbabf">

**Comment by yuvaltassa** [comment]
> when mjd_transitionFD is called, I don't need to call mj_step anymore, is this right?

No. `mjd_transitionFD` does not step the state.

Regarding your error, I think it is pretty clear. Please read the function [documentation](https://mujoco.readthedocs.io/en/latest/APIreference/APIfunctions.html#derivatives-api) and play with the [LQR colab](https://github.com/google-deepmind/mujoco#getting-started) which uses this function for clarification.

Finally [Please use issues rather than discussions](https://github.com/google-deepmind/mujoco#asking-questions). If you have more questions, open an issue.

**Reply by yzd0014** [reply]
Thanks for your reply. But I'm still very confused about what caused the error. The documentation says the dim for B matrix is 2*nv+na x nu, this is exactly what I did.

**Reply by yuvaltassa** [reply]
You are using an old version of MuJoCo, before [this commit](https://github.com/google-deepmind/mujoco/commit/e3cdc410d6b75e37995fe49c3f28e76384cac1fe) and there is a mismatch between the documentation and the code.

If you read the error it clearly says that it is expecting `centered` but you are passing `flg_centered`.

**Reply by yzd0014** [reply]
Thank you very much!
